sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/policy/ScrubEtagPolicy.java

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common.policy;
+
+import com.azure.core.http.HttpHeader;
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class ScrubEtagPolicy implements HttpPipelinePolicy {
+    private static final String ETAG = "eTag";
+
+    private HttpResponse innerHttpResponse;
+    private HttpHeaders headers;
+
+    /**
+     * Wraps any potential error responses from the service and applies post processing of the response's eTag header to
+     * standardize the value.
+     *
+     * @return an updated response with post processing steps applied.
+     */
+    @Override
+    public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        return next.process()
+            .flatMap(response -> {
```

Comment:
This could just be `.flatMap(response -> Mono.justOrEmpty(scrubETagHeader(response))`.

Also, do we need to have `justOrEmpty` as this shouldn't be able to be null.

Reply:
Done.